PowerShellTools/Intellisense/Utilities.cs

```diff
@@ -11,67 +11,77 @@ namespace PowerShellTools.Intellisense
 {
     internal static class Utilities
     {
-	internal static bool IsLeftBrace(char ch)
-	{
-	    return ch == '{' || ch == '[' || ch == '(';
-	}
-
-	internal static bool IsRightBrace(char ch)
-	{
-	    return ch == '}' || ch == ']' || ch == ')';
-	}
-
-	internal static bool IsLeftBraceOrQuotes(char ch)
-	{
-	    return IsLeftBrace(ch) || IsQuotes(ch);
-	}
-
-	internal static bool IsRightBraceOrQuotes(char ch)
-	{
-	    return IsRightBrace(ch) || IsQuotes(ch);
-	}
-
-	internal static bool IsQuotes(char ch)
-	{
-	    return ch == '\'' || ch == '\"';
-	}
-
-	internal static bool IsLeftCurlyBrace(char ch)
-	{
-	    return ch == '{';
-	}
-
-	internal static bool IsRightCurlyBrace(char ch)
-	{
-	    return ch == '}';
-	}
-
-	internal static char GetCloseBraceOrQuotes(char ch)
-	{
-	    switch (ch)
-	    {
-		case '{': return '}';
-		case '[': return ']';
-		case '(': return ')';
-		case '\'': return '\'';
-		case '\"': return '\"';
-		default: throw new InvalidOperationException("The character is unrecognized.");
-	    }
-	}
-
-	internal static char GetPairedBrace(char ch)
-	{
-	    switch (ch)
-	    {
-		case '{': return '}';
-		case '}': return '{';
-		case '[': return ']';
-		case ']': return '[';
-		case '(': return ')';
-		case ')': return '(';
-		default: throw new InvalidOperationException("The character is unrecognized.");
-	    }
-	}	
+        internal static bool IsGroupStart(char ch)
+        {
+            return ch == '{' || ch == '(';
+        }
+
+        internal static bool IsGroupEnd(char ch)
+        {
+            return ch == '}' || ch == ')';
+        }
+
+        internal static bool IsLeftBrace(char ch)
+        {
+            return ch == '{' || ch == '[' || ch == '(';
+        }
+
+        internal static bool IsRightBrace(char ch)
+        {
+            return ch == '}' || ch == ']' || ch == ')';
+        }
+
+        internal static bool IsLeftBraceOrQuotes(char ch)
+        {
+            return IsLeftBrace(ch) || IsQuotes(ch);
+        }
+
+        internal static bool IsRightBraceOrQuotes(char ch)
+        {
+            return IsRightBrace(ch) || IsQuotes(ch);
+        }
+
+        internal static bool IsQuotes(char ch)
+        {
+            return ch == '\'' || ch == '\"';
+        }
+
+        internal static bool IsLeftCurlyBrace(char ch)
+        {
+            return ch == '{';
+        }
+
+        internal static bool IsRightCurlyBrace(char ch)
+        {
+            return ch == '}';
+        }
+
+        internal static char GetCloseBraceOrQuotes(char ch)
+        {
+            switch (ch)
+            {
+                case '{': return '}';
+                case '(': return ')';
+                case '[': return ']';
+                case '\'': return '\'';
+                case '\"': return '\"';
+                default: throw new InvalidOperationException("The character is unrecognized.");

```

Comment:
nit: We shouldn't have a string in these two exceptions [eventual localization].  If you want to throw an exception here, throw new argument exception (ch..)